credentials/handler.go

```diff
@@ -0,0 +1,303 @@
+	if v.Type == identity.CredentialsTypeOIDC {
+		// oidc may contain more than one entry, so it has to be edited
+		var credentialsConfig oidc.CredentialsConfig
+		if err := json.Unmarshal(creds.Config, &credentialsConfig); err != nil {
+			return nil, err
+		}
+		var newProviders []oidc.ProviderCredentialsConfig
+		for _, val := range credentialsConfig.Providers {
+			if val.Provider != v.Filter {
+				newProviders = append(newProviders, val)
+			}
+		}
+		credentialsConfig.Providers = newProviders
+		if err := h.replaceConfig(creds, credentialsConfig); err != nil {
+			return nil, err
+		}
+		return creds, nil
+	} else if v.Type == identity.CredentialsTypePassword {
+		// password can just be removed
+		return nil, nil
+	} else {
+		return nil, errors.New("Illegal credentials type")
+	}
+}
+
+func (h *Handler) createCredentials(ctx context.Context, identityId uuid.UUID, newCreds ImportCredentialsBody) (*identity.Credentials, error) {
+	if newCreds.Type == identity.CredentialsTypeOIDC {
+		var newConfig oidc.CredentialsConfig
+		if err := json.Unmarshal(newCreds.Config, &newConfig); err != nil {
+			return nil, err
+		}
+		for _, provider := range newConfig.Providers {
+			if err := h.doesAnotherIdentityUseTheseOidcCredentials(ctx, identityId, provider); err != nil {
+				return nil, err
+			}
+		}
+		return h.newCredentials(newConfig, newCreds)
+	} else if newCreds.Type == identity.CredentialsTypePassword {
+		var newConfig password.CredentialsConfig
+		if err := json.Unmarshal(newCreds.Config, &newConfig); err != nil {
+			return nil, err
+		}
+		return h.newCredentials(newConfig, newCreds)
+	} else {
+		return nil, errors.New("Illegal credentials type")
+	}
+}
+
+func (h *Handler) newCredentials(newConfig interface{}, newCreds ImportCredentialsBody) (*identity.Credentials, error) {
+	var b bytes.Buffer
+	if err := json.NewEncoder(&b).Encode(newConfig); err != nil {
+		return nil, err
+	}
+	return &identity.Credentials{
+		Type:        newCreds.Type,
+		Identifiers: []string{},
+		Config:      b.Bytes(),
+	}, nil
+}
+
+func (h *Handler) updateCredentials(ctx context.Context, identityId uuid.UUID, oldCreds *identity.Credentials, newCreds ImportCredentialsBody) error {
+	if newCreds.Type == identity.CredentialsTypeOIDC {
+		// for OIDC - replace the ones in the import and leave all the rest w/o changing
+		var newConfig oidc.CredentialsConfig
+		if err := json.Unmarshal(newCreds.Config, &newConfig); err != nil {
+			return err
+		}
+		var oldConfig oidc.CredentialsConfig
+		if err := json.Unmarshal(oldCreds.Config, &oldConfig); err != nil {
+			return err
+		}
+		for _, provider := range newConfig.Providers {
+			if err := h.doesAnotherIdentityUseTheseOidcCredentials(ctx, identityId, provider); err != nil {
+				return err
+			}
+		}
+
+		var newProviders []oidc.ProviderCredentialsConfig
+		for _, oldV := range oldConfig.Providers {
+			added := false
+			for _, newV := range newConfig.Providers {
```

Comment:
We could iterate over `newConfig.Providers` first to avoid `for _, newV := range newConfig.Providers {` from 248 line